packages/suite/src/views/settings/SettingsCoins/SettingsCoins.tsx

```diff
@@ -81,18 +82,27 @@ const getDiscoveryButtonAnimationConfig = (isConfirmed: boolean): MotionProps =>
 
 export const SettingsCoins = () => {
     const { firmwareTypeBannerClosed } = useSelector(selectSuiteFlags);
-
     const isDiscoveryButtonVisible = useRediscoveryNeeded();
     const { mainnets, testnets, enabledNetworks, setEnabled } = useEnabledNetworks();
-    const deviceSupportedNetworks = useSelector(selectDeviceSupportedNetworks);
-    const supportedEnabledNetworks = enabledNetworks.filter(enabledNetwork =>
-        deviceSupportedNetworks.includes(enabledNetwork),
-    );
-
+    const deviceSupportedNetworkSymbols = useSelector(selectDeviceSupportedNetworks);
+    const deviceModel = useSelector(selectDeviceModel);
     const { device } = useDevice();
     const dispatch = useDispatch();
     const { isDiscoveryRunning } = useDiscovery();
 
+    const supportedEnabledNetworks = enabledNetworks.filter(enabledNetwork =>
+        deviceSupportedNetworkSymbols.includes(enabledNetwork),
+    );
+    const supportedNetworks = mainnets.filter(({ symbol }) =>
+        deviceSupportedNetworkSymbols.includes(symbol),
+    );
+    const unsupportedNetworks = mainnets.filter(
+        ({ symbol }) => !deviceSupportedNetworkSymbols.includes(symbol),
+    );
+    const supportedTestnetNetworks = testnets.filter(({ symbol }) =>
+        deviceSupportedNetworkSymbols.includes(symbol),
```

Comment:
Just thinking that the funcion inside is _almost_ the same in these 3 functions 🤔. Could we unify it? WDYT?

Reply:
Yeah I know. I actually wrote a complicated reduce to make it unified then decided it's too hard to read. But I could write a simple function:)